src/ol_infrastructure/lib/fastly.py

```diff
@@ -24,3 +24,23 @@ def get_fastly_provider(
     if wrap_in_pulumi_options:
         fastly_provider = pulumi.ResourceOptions(provider=fastly_provider)
     return fastly_provider
+
+
+fastly_log_format_string = """{
+    "timestamp": "%{strftime(\\{"%Y-%m-%dT%H:%M:%S%z"\\}, time.start)}V",
+    "client_ip": "%{req.http.Fastly-Client-IP}V",
+    "geo_country": "%{client.geo.country_name}V",
+    "geo_city": "%{client.geo.city}V",
+    "host": "%{if(req.http.Fastly-Orig-Host, req.http.Fastly-Orig-Host, req.http.Host)}V",
+    "url": "%{json.escape(req.url)}V",
+    "request_method": "%{json.escape(req.method)}V",
+    "request_protocol": "%{json.escape(req.proto)}V",
+    "request_referer": "%{json.escape(req.http.referer)}V",
+    "request_user_agent": "%{json.escape(req.http.User-Agent)}V",
+    "response_state": "%{json.escape(fastly_info.state)}V",
+    "response_status": %{resp.status}V,
+    "response_reason": %{if(resp.response, "%22"+json.escape(resp.response)+"%22", "null")}V,
+    "response_body_size": %{resp.body_bytes_written}V,
+    "fastly_server": "%{json.escape(server.identity)}V",
+    "fastly_is_edge": %{if(fastly.ff.visits_this_service == 0, "true", "false")}V
```

Comment:
Is it possible to get any information from Fastly on response time?

Reply:
oh let me look, I just stole the default one from manually setting it up in the UI.

Reply:
So, we actually have S3 logging set up on the legacy OCW distribution and the format string that was used there is:
```
{"timestamp": "%{%Y-%m-%dT%H:%M:%S+00:00}t", "client": "%h", "method": "%m", "request": "%{json.escape(req.url)}V", "status": "%>s", "header_bytes_sent": %{resp.header_bytes_written}V, "body_bytes_sent": %{resp.body_bytes_written}V, "referer": "%{json.escape(req.http.Referer)}V", "user_agent": "%{json.escape(req.http.User-Agent)}V", "request_time": %{time.elapsed}V, "time_to_first_byte": %{time.to_first_byte}V, "cache_status": "%{regsub(fastly_info.state, "^(HIT-(SYNTH)|(HITPASS|HIT|MISS|PASS|ERROR|PIPE)).*", "\\2\\3") }V", "city": "%{client.geo.city.utf8}V", "region": "%{client.geo.region.utf8}V", "country": "%{client.geo.country_name.utf8}V", "country_code": "%{client.geo.country_code}V", "continent_code": "%{client.geo.continent_code}V", "client_conn_type": "%{client.geo.conn_type}V", "client_geo_conn_speed": "%{client.geo.conn_speed}V"}
```
There are some interesting bits in there, so it's probably worth melding the two format strings to be a superset of all of this information so we can capture as much as possible.

Reply:
In general I like the field naming that you used for the attributes (e.g. `request_`, `response_`, `geo_`, etc. to make it clear what each attribute pertains to)

Reply:
Looks great to me

Reply:
Oh follow up question: would it hurt/hinder to have these compressed when they show up? I suspect it is going to be A LOT of data from production (60-100 rps). I can set a flag and they will show up gzipped instead of just raw json but I feel like that might add a complication to whatever future consumer of the data.

Reply:
GZIP is fine. The likely consumer is probably Trino which can handle compression without issue.